src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1630,6 +1654,61 @@ private void MakeSeparatorList(ReadOnlySpan<char> separators, ref ValueListBuild
             }
         }
 
+        private void MakeSeparatorListVectorized(ref ValueListBuilder<int> sepListBuilder, char c, char? c2 = null, char? c3 = null)
+        {
+            Vector256<byte> shuffleConstant = Vector256.Create(0x02, 0x06, 0x0A, 0x0E, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
+                                                        0xFF, 0xFF, 0xFF, 0xFF, 0x02, 0x06, 0x0A, 0x0E, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF);
+
+            Vector256<ushort> v1 = Vector256.Create(c);
+            Vector256<ushort>? v2 = c2 is char sep2 ? Vector256.Create(sep2) : (Vector256<ushort>?)null;
+            Vector256<ushort>? v3 = c3 is char sep3 ? Vector256.Create(sep3) : (Vector256<ushort>?)null;
+
+            ref char c0 = ref MemoryMarshal.GetReference(this.AsSpan());
+            int cond = Length - (Length % Vector256<ushort>.Count);
+            int i = 0;
+
+            for (; i < cond; i += Vector256<ushort>.Count)
+            {
+                ref char ri = ref Unsafe.Add(ref c0, i);
+                Vector256<ushort> charVector = Unsafe.As<char, Vector256<ushort>>(ref ri);
+                Vector256<ushort> cmp = Avx2.CompareEqual(charVector, v1);
+
+                if (v2 is Vector256<ushort> vecSep2)
+                {
+                    cmp = Avx2.Or(Avx2.CompareEqual(charVector, vecSep2), cmp);
+                }
+
+                if (v3 is Vector256<ushort> vecSep3)
+                {
+                    cmp = Avx2.Or(Avx2.CompareEqual(charVector, vecSep3), cmp);
+                }
+
+                if (Avx.TestZ(cmp, cmp)) { continue; }
+
+                Vector256<byte> mask = Avx2.ShiftLeftLogical(cmp.AsUInt64(), 4).AsByte();
+                mask = Avx2.Shuffle(mask, shuffleConstant);
+
+                Vector128<byte> res = Sse2.Or(Avx2.ExtractVector128(mask, 0), Avx2.ExtractVector128(mask, 1));
+                ulong extractedBits = Bmi2.X64.ParallelBitExtract(0xFEDCBA9876543210, Sse2.X64.ConvertToUInt64(res.AsUInt64()));
```

Comment:
I was thinking of moving this to a constant, but I feel like it would pollute the String class a little `0xFEDCBA9876543210`.
Should this be a local?

Reply:
It could be a local constant, if you like - then you could name it and maybe comment it.

Reply:
Oh my, I can't believe I didn't know that local constants existed in C#. I remember once looking up whether static locals existed, but that didn't seem to be the case. But this works, thank you ;)

Reply:
Adjusted in #https://github.com/dotnet/runtime/pull/38001/commits/52ad5acd1fe5dc4ecc830828b68077d70e05e757